src/cluster.c

```diff
@@ -98,7380 +49,1439 @@ static void clusterBuildMessageHdr(clusterMsg *hdr, int type, size_t msglen);
-         *    claims to serve slots we know are served by a master with a
-         *    greater configEpoch. If this happens we inform the sender.
-         *
-         * This is useful because sometimes after a partition heals, a
-         * reappearing master may be the last one to claim a given set of
-         * hash slots, but with a configuration that other instances know to
-         * be deprecated. Example:
-         *
-         * A and B are master and slave for slots 1,2,3.
-         * A is partitioned away, B gets promoted.
-         * B is partitioned away, and A returns available.
-         *
-         * Usually B would PING A publishing its set of served slots and its
-         * configEpoch, but because of the partition B can't inform A of the
-         * new configuration, so other nodes that have an updated table must
-         * do it. In this way A will stop to act as a master (or can try to
-         * failover if there are the conditions to win the election). */
-        if (sender && dirty_slots) {
-            int j;
-
-            for (j = 0; j < CLUSTER_SLOTS; j++) {
-                if (bitmapTestBit(hdr->myslots,j)) {
-                    if (server.cluster->slots[j] == sender ||
-                        server.cluster->slots[j] == NULL) continue;
-                    if (server.cluster->slots[j]->configEpoch >
-                        senderConfigEpoch)
-                    {
-                        serverLog(LL_VERBOSE,
-                            "Node %.40s has old slots configuration, sending "
-                            "an UPDATE message about %.40s",
-                                sender->name, server.cluster->slots[j]->name);
-                        clusterSendUpdate(sender->link,
-                            server.cluster->slots[j]);
-
-                        /* TODO: instead of exiting the loop send every other
-                         * UPDATE packet for other nodes that are the new owner
-                         * of sender's slots. */
-                        break;
-                    }
-                }
-            }
-        }
-
-        /* If our config epoch collides with the sender's try to fix
-         * the problem. */
-        if (sender &&
-            nodeIsMaster(myself) && nodeIsMaster(sender) &&
-            senderConfigEpoch == myself->configEpoch)
-        {
-            clusterHandleConfigEpochCollision(sender);
-        }
+    if (c->argc == 2 && !strcasecmp(c->argv[1]->ptr,"help")) {
+        clusterCommandHelp(c);
+    } else if (!strcasecmp(c->argv[1]->ptr,"myid") && c->argc == 2) {
+        /* CLUSTER MYID */
+        clusterCommandMyId(c);
+    } else if (!strcasecmp(c->argv[1]->ptr,"myshardid") && c->argc == 2) {
+        /* CLUSTER MYSHARDID */
+        clusterCommandMyShardId(c);
+    } else if (!strcasecmp(c->argv[1]->ptr,"slots") && c->argc == 2) {
+        /* CLUSTER SLOTS */
+        clusterReplySlots(c);
```

Comment:
If we're already doing this refactoring, we should update the .json files to directly point to clusterReplySlots instead of pointing to the container command.

Reply:
hmmm...why? I find it convenient that all the various subcommands are navigable from the same function

Reply:
@guybe7 This was the strategy we wanted to take right? I don't like that we have two conventions, one is to have a parent function with a bunch of sub-functions and one where each child function gets its own API. 

I don't really have a strong preference I just dislike inconsistency for the sake of it.